### DISCUSSION TITLE
Cannot find 'wasm' plugin

### DISCUSSION THREAD

**Discussion by Shamaoke** [radareorg/radare2#21712] [Q&A]
Hi.

I'm trying to open a WebAssembly binary file (`r2 sample.wasm`). Every time I open it the program outputs the following.

```
ERROR: Cannot find 'wasm' asm/arch/anal plugin. See rasm2 -L or -LL
ERROR: Cannot find 'wasm' asm/arch/anal plugin. See rasm2 -L or -LL
ERROR: Cannot find 'wasm' asm/arch/anal plugin. See rasm2 -L or -LL
```

What do the program want to inform me about in this case?

What actions should I perform in response to that information?

Thanks.

_radare2 5.8.4 0 @ linux-x86-64 git.5.8.4_
_commit: unknown build: 2023-04-28__20:51:49_

**Comment by trufae** [comment]
The wasm plugin is there and its tested in the ci. Which distro or how you build it?

**Comment by Shamaoke** [comment]
The distribution is Arch. I installed Radare2 from the official repository (`community/radare2`).

**Comment by trufae** [comment]
This is not the first issue i receive from bugs in the arch package that are not a problem in r2. Please report it there and install r2 from git or sources

**Comment by Shamaoke** [comment]
I compiled and installed Radare2 from [AUR](https://aur.archlinux.org/packages/radare2-git) and still get the same errors.

**Reply by trufae** [reply]
As expected. There are several things wrong in the aur package. Build r2 from git following the standard steps.

**Comment by Shamaoke** [comment]
@trufae Could you explain, what the exact problem with the [PKGBUILD](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=radare2-git) script? How the script can be improved?

I filed a [bug report](https://bugs.archlinux.org/task/78416) downstream. One of the commenters pointed out that the problem could be related to building with the Meson system. He also said that this [change](https://github.com/radareorg/radare2/commit/5d5a7fb6e308eb55c9909fab154d31f2eb2c91fe) could be the possible cause of the problem.

**Comment by trufae** [comment]
The wasm plugin for meson is optional and disabled by default. See libr/meson.build and grep for wasm

**Reply by antonio-rojas** [reply]
So, as far as I can see, specifying a list of plugins will override the default plugin list. What if one wants to *append* a plugin to the default list? (ie. build the default list + wasm)

**Reply by trufae** [reply]
Correct, when i finish the arch refactoring i may spend some time making all plugins optional and having the same defaults for meson and acr. actually i should run the ci with the meson build to catch those differences 👍

**Comment by trufae** [comment]
Other issues i find is that is linking against cs4 which is like 5 year old and it misses so many instructions that it makes it useless for any modern x64 or arm64 binary

**Comment by trufae** [comment]
The url must be radareorg not radare.
Openssl is only necessary for the tls support which basically nobody uses.
libuv is totally not necessary and support is not well tested. Same for libmagic.

i dont get the touch libr/config.mk line either

The pkgversion can be just ./configure -qV

**Comment by trufae** [comment]
The r2.1 manpage should be installed in the latest release. Not sure about the doc, but this should be fixed in my side instead of manually copying the files

**Comment by trufae** [comment]
Thanks for reporting and taking care of this :) i really hope arch gets better builds of r2

**Comment by Shamaoke** [comment]
Thank you for your kindly help!

**Comment by trufae** [comment]
wasm is now included by default in meson builds :) thanks for reporting!
